slackbot/bot.py

```diff
@@ -37,6 +37,9 @@ def _keepactive(self):
         logger.info('keep active thread started')
         while True:
             time.sleep(30 * 60)
+            users = dict((u['id'], u) for u in self._client.webapi.users.list()['body']['members'])
+            if users != self._client.users:
+                self._client.users = users
```

Comment:
Wrap it in a `try .. except` block (and log the exception) so that unexpected http errors won't break the whole app.